LinqToRest.Test/QueryProviderTest.cs

```diff
@@ -13,66 +14,209 @@ public class QueryProviderTest
-            return new UriBuilder(root).Uri;
+            var actual = new QueryResult<EntityWithQueryableMember>(
+                CreateQuery<EntityWithQueryableMember>());
+
+            var expected = EntityWithQueryableMemberResult;
+
+            Assert.Equal(expected, actual);
         }
 
-        private static IResourceRetriever MockResourceRetriever()
+        [Fact]
+        public void ReturnsRestObjectWithSelect()
         {
-            var resourceRetriever = Substitute.For<IResourceRetriever>();
+            var actual = new QueryResult<object>(
+                CreateQuery<EntityWithQueryableMember>()
+                    .Select(entity => new { entity.Name }));
 
-            var uri = new Uri(MockServiceUri(), "test");
-            var result = new[]
-            {
-                new EntityWithQueryableMember("Test1", null),
-                new EntityWithQueryableMember("Test2", null)
-            };
-            resourceRetriever.RetrieveResource<IEnumerable<EntityWithQueryableMember>>(uri).Returns(result);
+            var expected = NameResult;
+
+            Assert.Equal(expected, actual);
+        }
+
+        [Fact]
+        public void ReturnsRestObjectWithSelectedUniqueIdentifier()
+        {
+            var actual = new QueryResult<object>(
+                CreateQuery<EntityWithQueryableMember>()
+                    .Select(entity => new { entity.UniqueIdentifier, entity.Name }));
 
-            return resourceRetriever;
+            var expected = UniqueIdentifierNameResult;
+
+            Assert.Equal(expected, actual);
         }
 
-        private static IObjectResolver MockObjectResolver()
+        #region Helper
+
+        private static IQueryable<T> CreateQuery<T>()
         {
-            var queryableFactory = new QueryableFactory(MockQueryProvider());
+            return new Query<T>(CreateQueryProvider(RootUri));
+        }
 
-            return new QueryableObjectResolver(queryableFactory);
+        private static IQueryable<T> CreateQuery<T>(string subPath)
+        {
+            return new Query<T>(CreateQueryProvider(new Uri(RootUri, subPath)));
         }
 
-        private static QueryProvider MockQueryProvider()
+        private static QueryProvider CreateQueryProvider(Uri root)
         {
-            return new QueryProvider(
-                MockResourceRetriever(),
-                MockQueryBinderFactory(),
-                MockServiceUri());
+            return new QueryProvider(CreateResourceRetriever(), null, () => new QueryBinder(new EntityValidator()), root);
         }
 
-        private static QueryBinderFactory MockQueryBinderFactory()
+        #endregion
+
+        #region Mock
+
+        private static HttpClient MockHttpClient()
         {
-            var queryBinderFactory = Substitute.For<QueryBinderFactory>();
-            queryBinderFactory().Returns(new QueryBinder(new EntityValidator()));
+            var mockHttp = new MockHttpMessageHandler();

```

Comment:
use `HttpClientMock()`

Reply:
Addressed in https://github.com/messerli-informatik-ag/linq-to-rest/pull/10/commits/f09a88cdbf86a71763697a2facc065dc21ddd2e3

Reply:
`retrun new HttpClientMock()
    .RegisterJasonResponse(...)
    .ToHttpClient();`

how about that?

Reply:
Addressed in https://github.com/messerli-informatik-ag/linq-to-rest/pull/10/commits/ae816f8de380d69c5b5467919bd74346e54c6310